crates/provider/src/provider/helper.rs

```diff
@@ -0,0 +1,30 @@
+use crate::{Provider, RootProvider};
+
+use alloy_network::Network;
+use std::sync::Arc;
+
+/// A wrapper struct around Arc dyn provider
+pub struct WrappedProvider<'a, N>(Arc<dyn Provider<N> + 'a>);
+
+impl<'a, N: Network> WrappedProvider<'a, N> {
+    pub fn new<P>(provider: &'a P) -> Self
```

Comment:
this should consume the provider because we don't want to use references for this

Reply:
Yup, If I do that, then the wrapped implementation would need me to constrain Clone on provider

```
   fn wrapped(&self) -> WrappedProvider<N>
    where
        Self: Sized + Clone + 'static,
    {
        WrappedProvider::new(self.clone())
    }
``` 

and that seems to break a few other things in the codebase as well, in contract crate and here https://github.com/Dhruv-2003/alloy/blob/045371819661cba7c2b5ebb9509a38d2c8ec58fe/crates/provider/src/fillers/nonce.rs#L175

Let me know if another approach is to be taken here